File: HomeWorks/HomeWorks009/Triangle.java
package HomeWorks.HomeWorks009;

import java.util.ArrayList;
import java.util.Arrays;

public class Triangle extends Polygon{
    public Triangle(double sideA, double sideB,double sideC) throws IllegalArgumentException{
        super(new ArrayList <> (Arrays.asList(sideA,sideB,sideC)));
        if (sideA+sideB<sideC && sideC+sideB<sideA && sideC+sideA<sideB) throw new IllegalArgumentException();
    }

    @Override
    double area() {
        double halfPerimeter = this.sides.get(0)+this.sides.get(1)+this.sides.get(2);
        return Math.sqrt(halfPerimeter*(halfPerimeter - this.sides.get(0))*(halfPerimeter - this.sides.get(1))*(halfPerimeter - this.sides.get(2)));
    }
}

File: HomeWorks/HomeWorks013/Core/Models/Triangle.py
from Core.Models.Polygon import Polygon


class Triangle(Polygon):
    def __init__(self, a, b, c):
        if a + b <= c or a + c <= b or b + c <= a:
            raise ValueError("Invalid triangle sides")
        super().__init__([a, b, c])

    def area(self):
        s = (self.sides[0] + self.sides[1] + self.sides[2]) / 2
        return (s * (s - self.sides[0]) * (s - self.sides[1]) * (s - self.sides[2])) ** 0.5
